src/methods/SANA.cpp

```diff
@@ -2865,14 +2868,24 @@ void SANA::initIterPerSecond() {
     initializedIterPerSecond = true;
 
     cout << "Determining iteration speed...." << endl;
-    long long int iter = 1E6;
-    hillClimbingIterations(iter - 1);
-    double res = iter/timer.elapsed();
-    cout << "SANA does " << to_string(res)
-         << " iterations per second (took " << timer.elapsedString()
-         << " doing " << iter << " iterations)" << endl;
-
-    iterPerSecond = res;
+    //long long int iter = 1E6;
+    //hillClimbingIterations(iter - 1);
+    //double res = iter/timer.elapsed();
+    //cout << "SANA does " << to_string(res)
+    //     << " iterations per second (took " << timer.elapsedString()
+    //     << " doing " << iter << " iterations)" << endl;
+    double totalIps = 0.0;
+    int ipsListSize = 0;
+    for(pair<double,double> ipsPair : ipsList){
+	if(TInitial>ipsPair.first && ipsPair.first>TFinal){
```

Comment:
I prefer the format a <= x <= b rather than b>=x>=a; also please add "equal", not just < or >.

Also, summing all the iters and all the times will give a biased outcome because at some temperatures it takes longer than others. Thus, create just *one* number at each temperature, which is the ips. Then just average all the ips's to get the final estimate.